client/tokenized-payment-request/cart-api.js

```diff
@@ -0,0 +1,131 @@
+	// This functionality is also useful to calculate product/shipping pricing (and shipping needs)
+	// for compatibility scenarios with other plugins (like WC Bookings, Product Add-Ons, WC Deposits, etc.).
+	cartRequestHeaders = {};
+
+	async placeOrder( paymentData ) {
+		return await apiFetch( {
+			method: 'POST',
+			path: '/wc/store/v1/checkout',
+			credentials: 'omit',
+			headers: {
+				'X-WC-Payments-prb-request': true,
+				...this.cartRequestHeaders,
+			},
+			data: paymentData,
+		} );
+	}
+
+	async getCart() {
+		return await apiFetch( {
+			method: 'GET',
+			path: '/wc/store/v1/cart',
+		} );
+	}
+
+	async createAnonymousCart() {
+		const response = await apiFetch( {
+			method: 'GET',
+			path: '/wc/store/v1/cart',
+			// omitting credentials, to create a new cart object separate from the user's cart.
+			credentials: 'omit',
+			// parse: false to ensure we can get the response headers
+			parse: false,
+		} );
+
+		this.cartRequestHeaders = {
+			Nonce: response.headers.get( 'Nonce' ),
+			'Cart-Token': response.headers.get( 'Cart-Token' ),
+		};
+	}
+
+	async updateCustomer( customerData ) {
+		return apiFetch( {
+			method: 'POST',
+			path: '/wc/store/v1/cart/update-customer',
+			credentials: 'omit',
+			headers: {
+				'X-WC-Payments-prb-request': true,
+				...this.cartRequestHeaders,
+			},
+			data: customerData,
+		} );
+	}
+	async selectShippingRate( shippingRate ) {
+		return apiFetch( {
+			method: 'POST',
+			path: '/wc/store/v1/cart/select-shipping-rate',
+			credentials: 'omit',
+			headers: {
+				...this.cartRequestHeaders,
+			},
+			data: shippingRate,
+		} );
+	}
+
+	async addProductToCart() {
+		const productData = {
+			// can be modified in case of variable products, WC bookings plugin, etc.
+			id: jQuery( '.single_add_to_cart_button' ).val(),
+			quantity: parseInt( jQuery( '.quantity .qty' ).val(), 10 ) || 1,
+			// can be modified in case of variable products, WC bookings plugin, etc.
+			variation: [],
+		};
+
+		return apiFetch( {
+			method: 'POST',
+			path: '/wc/store/v1/cart/add-item',
+			credentials: 'omit',
+			headers: {
+				...this.cartRequestHeaders,
+			},
+			data: applyFilters(
+				'wcpay.payment-request.cart-add-item',
+				productData
+			),
+		} );
+	}
+
+	async emptyCart() {
+		try {
+			const cartData = await apiFetch( {
+				method: 'GET',
+				path: '/wc/store/v1/cart',
+				credentials: 'omit',
+				headers: {
+					...this.cartRequestHeaders,
+				},
+			} );
+
+			const removeItemsPromises = cartData.items.map( ( item ) => {
+				return apiFetch( {
+					method: 'GET',
+					path: '/wc/store/v1/cart',
+					credentials: 'omit',
+					headers: {
+						...this.cartRequestHeaders,
+					},
+					data: {
+						key: item.key,
+					},
+				} );
+			} );
+
+			await Promise.all( removeItemsPromises );
+		} catch ( e ) {
+			// let's ignore the error, it's likely not going to be relevant. We can just clean the cached headers.
+		}
+
+		this.cartRequestHeaders = {};
```

Comment:
```suggestion
		this.cartRequestHeaders = {};
```
At first, I thought it was irritating that the Store API would not provide us a function to empty the cart...but then when I thought about this further, is this clean-up something that we even need to be concerned about at all? 🤔 

Is there a need to actually physically empty the contents of the cart once we are finished with it? AFAICT, in the single instance I can see where we use this `emptyCart()` function, we just throw away the `temporaryCart` object once we are finished with it anyways, so I can't see an obvious way for us to recall the contents of this expired cart in any case. Let me know if I'm missing anything--which I probably am--but do we ever really even _need_ to call this `emptyCart()` function at all?

Reply:
That's a good question - I added the "clearing" function to support scenarios where a product with "limited availability" is added to the cart - something similar to what happened here: https://github.com/Automattic/woocommerce-payments/issues/7256#issuecomment-1813245817

Basically, I am afraid that if a product with stock of "1" is added to the cart (and the cart is not cleared), other customers might not be able to check out the product if the store has PRBs enabled.

I added a comment in the code to clarify 👍